oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java

```diff
@@ -157,6 +157,10 @@ public class MongoDocumentStore implements DocumentStore {
      */
     public static final long DEFAULT_THROTTLING_TIME_MS = Long.getLong("oak.mongo.throttlingTime", 20);
 
+    /**
+     * Document size of 16MB is a limit in Mongo
+     */
+    public static final long SIZE_LIMIT = 16793600;
```

Comment:
This is a bit larger than 16MB, what's the reason? I was also wondering whether we shouldn't warn below 16MB already to have some buffer, wdyt?

Reply:
> This is a bit larger than 16MB, what's the reason? I was also wondering whether we shouldn't warn below 16MB already to have some buffer, wdyt?

Actually, Mongo compares with this size and throws an exception. IMO, it would be good to give a WARN when the size reaches 15MB may be. wdyt?

Reply:
> I think keeping 15MB as warning would be good

+1

Reply:
Having said that, as @rishabhdaim  pointed out below, the updateOp might not be the one throwing the document over the 16MB and going via catching the exception might be the only reliable way in the end (in which case we probably don't event need these checks here

Reply:
yes adding try/catch only